modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache

```diff
@@ -223,15 +223,15 @@ public class {{classname}} {{#parent}}extends {{{.}}}{{/parent}}{{^parent}}{{#ha
   public {{>nullableDataType}} {{getter}}() {
   {{/isNullable}}
   {{^isNullable}}
-  public {{#isMap}}{{>nullableDataType}}{{/isMap}}{{#isArray}}{{>nullableDataType}}{{/isArray}}{{^isMap}}{{^isArray}}{{#required}}{{>nullableDataType}}{{/required}}{{^required}}{{>optionalDataTypeWithEnum}}{{/required}}{{/isArray}}{{/isMap}} {{getter}}() {
+  public {{>dataTypeDeclaration}} {{getter}}() {
```

Comment:
Looks good!

Reply:
I think this can still be improved, but it'll be a lot easier with https://github.com/OpenAPITools/openapi-generator/pull/11666/files so I'll iterate on it after that

Reply:
Did you want to get https://github.com/OpenAPITools/openapi-generator/pull/11666 in first? This change is good to go otherwise, but might be easier to deal with merge conflicts and resolve any unit tests changes if I update this after yours is in first.